packages/flutter_tools/test/linux/linux_device_test.dart

```diff
@@ -59,6 +61,27 @@ void main() {
       Platform: () => notLinux,
     });
   });
+
+  testUsingContext('LinuxDevice.isSupportedForProject is true with editable host app', () async {
+    fs.file('pubspec.yaml').createSync();
+    fs.file('.packages').createSync();
+    fs.directory('linux').createSync();
+    final FlutterProject flutterProject = await FlutterProject.current();
+
+    expect(LinuxDevice().isSupportedForProject(flutterProject), true);
+  }, overrides: <Type, Generator>{
+    FileSystem: () => MemoryFileSystem(),
+  });
+
+  testUsingContext('LinuxDevice.isSupportedForProject is false with no host app', () async {
+    fs.file('pubspec.yaml').createSync();
+    fs.file('.packages').createSync();
+    final FlutterProject flutterProject = await FlutterProject.current();
+
+    expect(LinuxDevice().isSupportedForProject(flutterProject), false);
```

Comment:
Is this still false when the pubspec.yaml is non-empty? That is, why does linux not have a test with non-empty pubspec.yaml like Android and iOS?

Reply:
linux, macos, and windows don't support the module style package (and probably won't ever)